### PR TITLE
Clearer docs for how to setup machine

### PR DESCRIPTION
**Pull request by jumaffre**
Resolves https://github.com/microsoft/CCF/issues/483

Update docs to make it clearer how to setup CCF:
- Point to WSL/Azure Cloud Shell for Windows setup
- Add a `pre_make_vm.sh` to install Azure CLI and oe-engine
- SSH Public Key to pass to `./make_vm.sh`

Also clarify concepts.

### CODE REVIEW

File: sphinx/source/quickstart/requirements.rst [hunk 1]
```diff
@@ -43,76 +34,44 @@ The CCF repository also provides a sample `devcontainer.json`_ file which will b
...
 
-3. Generate Azure Resource Manager deployment templates. This assumes that you are using an `SSH key`_ to authenticate, but it is also possible to use a password with adminPassword_.
+Then, to quickly get a VM up and running (in the East US region), you can run the following command, specifying your personal Azure subscription as environment variable:
 
 .. code-block:: bash
 
-    $ oe-engine generate --api-model vm.json --ssh-public-key ~/.ssh/id_rsa.pub --output-directory vm
+    $ SUBSCRIPTION=$AZURE_SUBSCRIPTION_NAME ./make_vm.sh [path_to_ssh_public_key]
+
+After signing in to your Azure account, the script will create a default ``ccf`` user on the VM, authenticated by the public key specified by ``path_to_ssh_public_key`` (defaults to ``~/.ssh/id_rsa.pub``). See :ref:`OE Engine Walkthrough` for further details about how to deploy ACC VM.
```
Comment (eddyashton) [inline sphinx/source/quickstart/requirements.rst]: "deploy ACC VMs" or "deploy an ACC VM"?

File: getting_started/create_vm/make_vm.sh [hunk 2]
```diff
@@ -10,6 +10,11 @@ RG=ccf-dev
 REGION=eastus
 CFG=vm.json
 
+# By default, use current ssh public key
```
Comment (olgavrou) [inline getting_started/create_vm/make_vm.sh]: it's always good for a script to print out any default's it is using

Comment (jumaffre) [reply getting_started/create_vm/make_vm.sh]: It should print it already (`set -x` at the top of the file).

File: sphinx/source/quickstart/oeengine.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
+OE Engine Walkthrough
+=====================
+
+:term:`OpenEnclave Engine` (OE Engine) offers detailed `deployment instructions`_, but this is a very condensed summary to get a CCF-ready VM up and running in 5 minutes. You can either execute these steps on a machine with the `Azure CLI`_ installed, or use `Azure Cloud Shell`_.
+
```
Comment (olgavrou) [inline sphinx/source/quickstart/oeengine.rst]: maybe add a note here saying that the pre_vm_make.sh script will install the Azure CLI and the oe-engine binary?

**Pull request merged**